spec.bs

```diff
@@ -3711,47 +3737,58 @@ algorithms to achieve the outcomes described in the above explanatory content.
 </div>
 
 <div algorithm>
-  Create a new algorithm, called <dfn>Derive a permissions policy directly from a fenced frame
-  config instance</dfn>.
+  Create a new algorithm, called <dfn>Create a permissions policy for a fenced navigable</dfn>.
 
-  Given null or an [=element=] (|container|), this algorithm returns a new [=permissions policy=].
+  Given an [=element=] (|container|) and an [=origin=] (|origin|), this algorithm returns a
+  new [=Permissions Policy=].
 
-  1. Let |inherited policy| be a new [=ordered map=].
+  1. [=Assert=]: |container| is a [=fenced navigable container=].
 
-  1. Let |effective permissions| be an empty [=list=].
-  
   1. Let |fenced frame config| be |container|'s [=Node/node document=]'s [=navigable/active browsing
      context=]'s [=browsing context/fenced frame config instance=].
 
-  1. If |fenced frame config| is not null, and |fenced frame config|'s [=fenced frame config/
-     effective enabled permissions=] are not null, set |effective permissions| to
-     |fenced frame config|'s [=fenced frame config/effective enabled permissions=].
+  1. Let |inherited policy| be a new [=ordered map=].
+
+  1. If |fenced frame config| is not null and |fenced frame config|'s [=fenced frame config
+     instance/permissions policy behavior=] is [=permissions policy behavior/fixed=], then:
+
+    1. [=list/For each=] |feature| [=supported features|supported=]:
+
+        1. If |fenced frame config|'s [=fenced frame config instance/effective enabled permissions=]
+           [=list/contains=] |feature|, then set |inherited policy|[feature] to "`Enabled`".
+
+           Otherwise, set |inherited policy|[feature] to "`Disabled`".
 
-  1. [=list/For each=] |feature| [=supported features|supported=]:
+        Note: While this doesn't take the <{fencedframe/allow}> attribute into consideration, it
+        will have already been checked by the time this is called because of [=Should navigation
+        response to navigation request be blocked by Permissions Policy?=]. Any policy specified in
+        <{fencedframe/allow}> that is too restrictive would have cause the fenced frame to not load,
+        and any policy that is more permissive than what is specified in the [=fenced frame
+        config/effective enabled permissions=] will be ignored.
 
-    1. If |effective permissions| [=list/contains=] |feature|, then set |inherited policy|[feature]
-       to "`Enabled`".
-       
-       Otherwise, set |inherited policy|[feature] to "`Disabled`".
+  1. Otherwise:
+
+    1. [=list/For each=] |feature| [=supported features|supported=]:
 
+      1. If |feature| matches one of the [=fenced frame allowed permissions=], then set
+        |inherited policy|[feature] to the result of running [$Define an inherited policy for
+        feature in container at origin$] given |feature|, |container|, and |origin|.
```

Comment:
We're intentionally not passing in `|fenced|` here, and that's correct, right?

Reply:
Correct. `|fenced|`/`|match all|` is only needed when determining whether to block navigation. In this instance, we want "flexible" permissions fenced frames to behave the same as their iframe counterparts (for permissions we allow to be enabled), so we want `|match all|` to be set to their default "false" value.